scripts/pre_commit_linter.py

```diff
@@ -666,15 +668,53 @@ def _check_import_order(all_files):
 
     return summary_messages
 
+
+def _check_def_spacing(all_files):
+    """This function is used to check the number of blank lines
+    left above each class, function and method defintion.
```

Comment:
`This function checks the number of blank lines above each class, function and method definition.`